manage-data/ingest/transform-enrich/error-handling.md

```diff
@@ -0,0 +1,153 @@
+      {
+        "set": {
+          "field": "event.kind",
+          "value": "pipeline_error"
+        }
+      },
+      {
+        "append": {
+          "field": "error.message",
+          "value": "Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message: {{ _ingest.on_failure_message }}"
+        }
+      }
+    ]
+  }
+}
+```
+
+The second processor, which sets `event.category` to `authentication`, is no longer executed because the first `dissect` processor fails and triggers the global `on_failure` handler. The resulting document shows which processor caused the error, the pattern it attempted to apply, and the input it received.
+
+```json
+"@timestamp": "2025-04-03T10:00:00.000Z",
+"message": "user: philipp has logged in",
+"event": {
+    "kind": "pipeline_error"
+},
+"error": {
+  "message": "Processor dissect with tag dissect for user.name in pipeline _simulate_pipeline failed with message: Unable to find match for dissect pattern: %{}, %{user.name} %{} against source: user: philipp has logged in"
+}
+```
+
+We can restructure the pipeline by moving the `on_failure` handling directly into the processor itself. This allows the pipeline to continue execution. In this case, the `event.category` processor still runs. You can also retain the global `on_failure` to handle errors from other processors, while adding processor-specific error handling where needed.
+
+(While executing two `set` processors within the `dissect` error handler may not always be ideal, it serves as a demonstration.)
+
+For the `dissect` processor, consider setting a temporary field like `_tmp.error: dissect_failure`. You can then use `if` conditions in later processors to execute them only if parsing failed, allowing for more controlled and flexible error handling.
+
+```json
+POST _ingest/pipeline/_simulate
+{
+  "docs": [
+    {
+      "_source": {
+        "@timestamp": "2025-04-03T10:00:00.000Z",
+        "message": "user: philipp has logged in"
+      }
+    }
+  ],
+  "pipeline": {
+    "processors": [
+      {
+        "dissect": {
+          "field": "message",
+          "pattern": "%{}, %{user.name} %{}",
+          "on_failure": [
+            {
+              "set": {
+                "field": "event.kind",
+                "value": "pipeline_error"
+              }
+            },
+            {
+              "append": {
+                "field": "error.message",
+                "value": "Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message: {{ _ingest.on_failure_message }}"
+              }
+            }
+          ],
+          "tag": "dissect for user.name"
+        }
+      },
+      {
+        "append": {
+          "field": "event.category",
+          "value": "authentication"
+        }
+      }
+    ],
+    "on_failure": [
```

Comment:
Feel free to ignore, but IME when users want global error handling they also want to [reroute](https://www.elastic.co/docs/reference/enrich-processor/reroute-processor) documents to a different failure index which might be nice to include.

Reply:
I wouldn't mention this anymore and just wait for the failure stream feature coming out at some point in time, and then we have people who manually implemented it and our out-of-the-box solution which is then conflicting with each other.